packages/web/components/swap-tool/price-selector.tsx

```diff
@@ -0,0 +1,490 @@
+import { Menu, Transition } from "@headlessui/react";
+import { Dec, PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY, MaybeUserAssetCoin } from "@osmosis-labs/server";
+import { Asset } from "@osmosis-labs/types";
+import { getAssetFromAssetList } from "@osmosis-labs/utils";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import { parseAsBoolean, parseAsString, useQueryState } from "nuqs";
+import React, { Fragment, memo, useEffect, useMemo } from "react";
+
+import { Icon } from "~/components/assets";
+import { AssetLists } from "~/config/generated/asset-lists";
+import { useDisclosure, useTranslation } from "~/hooks";
+import { useOrderbookSelectableDenoms } from "~/hooks/limit-orders/use-orderbook";
+import { AddFundsModal } from "~/modals/add-funds";
+import { useStore } from "~/stores";
+import { formatFiatPrice } from "~/utils/formatter";
+import { api } from "~/utils/trpc";
+
+type AssetWithBalance = Asset & MaybeUserAssetCoin;
+
+const UI_DEFAULT_QUOTES = ["USDC", "USDT"];
+
+const VALID_QUOTES = [
+  ...UI_DEFAULT_QUOTES,
+  "USDC.sol.axl",
+  "USDC.sol.wh",
+  "USDC.eth.grv",
+  "USDC.eth.wh",
+  "USDC.matic.axl",
+  "USDC.avax.axl",
+  "USDC.eth.axl",
+  "USDT.sol.axl",
+  "USDT.eth.grv",
+  "USDT.eth.wh",
+  "USDT.matic.axl",
+  "USDT.avax.axl",
+  "USDT.kava",
+  "USDT.eth.pica",
+  "USDT.sol.pica",
```

Comment:
I see a lot of usage of denoms as keys. Note, these are subject to chainge often and for that reason chain denoms (ibc hashes) would be safer. Up to you to determine if that is a risk here.

Reply:
Is this caught by Jose's suggestion to use `MainnetAssetSymbol`?